tests/tck-build-logic/src/main/java/org/graalvm/internal/tck/ContributionTask.java

```diff
@@ -0,0 +1,500 @@
+    }
+
+    private void loadQuestions() throws IOException {
+        File questionsJson = getProject().file("tests/tck-build-logic/src/main/resources/contributing/questions.json");
+        List<Question> contributingQuestions = objectMapper.readValue(questionsJson, new TypeReference<>() {});
+        for (var question : contributingQuestions) {
+            this.questions.put(question.questionKey(), new ContributingQuestion(question.question(), question.help()));
+        }
+    }
+
+    @TaskAction
+    void run() throws IOException {
+        InteractiveTaskUtils.printUserInfo("Hello! This task will help you contributing to metadata repository." +
+                " Please answer the following questions. In case you don't know the answer on the question, type \"help\" for more information");
+
+        loadQuestions();
+
+        this.coordinates = getCoordinates();
+        InteractiveTaskUtils.closeSection();
+
+        Path coordinatesMetadataRoot = getProject().file(CoordinateUtils.replace("metadata/$group$/$artifact$", coordinates)).toPath();
+        boolean isExistingLibrary = Files.exists(coordinatesMetadataRoot);
+
+        Path testsLocation = getTestsLocation();
+        InteractiveTaskUtils.closeSection();
+
+        Path resourcesLocation = getResourcesLocation();
+        InteractiveTaskUtils.closeSection();
+
+        List<String> dockerImages = getDockerImages();
+        InteractiveTaskUtils.closeSection();
+
+        List<String> packages = getAllowedPackages();
+        InteractiveTaskUtils.closeSection();
+
+        List<Coordinates> additionalTestImplementationDependencies = getAdditionalDependencies();
+        InteractiveTaskUtils.closeSection();
+
+        // initialize project
+        initializeWorkingDirectories();
+        createStubs(isExistingLibrary);
+        updateAllowedPackages(packages, isExistingLibrary);
+
+        // generate necessary infrastructure
+        addTests(testsLocation);
+        addResources(resourcesLocation);
+        addDockerImages(dockerImages);
+        addUserCodeFilterFile(packages);
+        addAdditionalDependencies(additionalTestImplementationDependencies);
+        addAgentConfigBlock();
+
+        // run agent in conditional mode
+        collectMetadata();
+
+        // create a PR
+        boolean shouldCreatePR = shouldCreatePullRequest();
+        if (shouldCreatePR) {
+            String branch = "add-support-for-" + coordinates.toString().replace(':', '-');
+            createPullRequest(branch);
+
+            InteractiveTaskUtils.printUserInfo("After your pull requests gets generated, please update the pull request description to mention all places where your pull request" +
+                    "accesses files, network, docker, or any other external service, and check if all checks in the description are correctly marked");
+        }
+
+        InteractiveTaskUtils.printSuccessfulStatement("Contribution successfully completed! Thank you!");
+    }
```

Comment:
Should have the whole questionnaire in a try-catch block?
Do you need to clean up if any of those `IllegalStateException`s below occur?

Reply:
I refactored code so that now we don't have anything that is not `ContributingException` or `RuntimeException`, so I guess there is no need for: 
>Should you have the whole questionnaire in a try-catch block?